apis/python/test/test_index.py

```diff
@@ -112,9 +116,9 @@ def test_index_with_incorrect_dimensions(tmp_path):
         # Okay otherwise.
         index.query(np.array([[1, 1, 1]], dtype=np.float32), k=3)
 
-def test_index_with_incorrect_num_of_query_columns_simple(tmp_path):
-    siftsmall_uri = "test/data/siftsmall/siftsmall_base.fvecs"
-    queries_uri = "test/data/siftsmall/siftsmall_query.fvecs"
+def test_index_with_incorrect_num_of_query_columns_simple(tmp_path, test_data_path):
+    siftsmall_uri = f"{test_data_path}/siftsmall/siftsmall_base.fvecs"
```

Comment:
For these we should use a variable.  In PR #173 (sorry for overloading that with so many changes) to make sure the testing is consistent wrt to file naming I created a file test/array_paths.py that defined variables for most of the test arrays and files:
[array_paths.txt](https://github.com/TileDB-Inc/TileDB-Vector-Search/files/13953198/array_paths.txt)

We should use that convention (both in terms of using variables rather than hard-coded names, as well as putting those into test/array_defs.py).

There are many duplications of this below -- this comment applies to each of them (I won't comment on each of them).